sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultTaskSpec.java

```diff
@@ -109,6 +82,32 @@ private DefaultTaskSpec(Builder builder) {
                 builder.discoverySpec,
                 builder.taskKillGracePeriodSeconds,
                 builder.transportEncryption);
+        ValidationUtils.nonEmpty(this, "name", name);
+        ValidationUtils.nonNull(this, "goalState", goalState);
+        ValidationUtils.nonNull(this, "essential", essential);
+        ValidationUtils.nonNull(this, "resourceSet", resourceSet);
+
+        // Check for duplicate names and paths:
+        ValidationUtils.atLeast(
```

Comment:
A `isUnique` check may reduce the boiler plate a little (especially if it accepts a stream).

Reply:
Created, SO had a way to do it efficiently from a stream as well